Lib/fontTools/ttLib/tables/_p_o_s_t.py

```diff
@@ -252,14 +255,34 @@ def fromXML(self, name, attrs, content, ttFont):
 			self.data = readHex(content)
 
 
-def unpackPStrings(data):
+def unpackPStrings(data, n):
+        # extract n Pascal strings from data.
+        # if there is not enough data, use ""
+
 	strings = []
 	index = 0
 	dataLen = len(data)
-	while index < dataLen:
-		length = byteord(data[index])
-		strings.append(tostr(data[index+1:index+1+length], encoding="latin1"))
-		index = index + 1 + length
+
+	for _ in range(n):
+		if (dataLen <= index):
+			length = 0
+		else:
+			length = byteord(data[index])
+		index += 1
+
+		if (dataLen <= index + length - 1):
+			name = ""
+		else:
+			name = tostr(data[index:index+length], encoding="latin1")
```

Comment:
if you try accessing a slice of a string which is outside its bounds, you already get an empty string (or a portion of the string that interesect the slice), so the `if` statement above is unnecessary, e.g.

```python
>>> s = "abc"
>>> s[2:6]
'c'
>>> s[3:4]
''
>>> s[4:5]
''
```

Reply:
The "if" part handles the case where the length of the string is n but we have < n bytes following it. We could use those bytes as the name, but since the string is not well formed, using "" seems better.